src/middlewares/isAuth.js

```diff
@@ -0,0 +1,17 @@
+const jwt = require('jsonwebtoken');
+
+module.exports = async (req, res, next) => {
+  const { token } = req.cookies;
+  try {
+    const { user } = jwt.verify(token, process.env.JWT_SECRET);
```

Comment:
yes, I was getting the user without deconstruction, but it was returning undefined because the verify function returns an object that contains the user object along with some properties like expiration time that's why I used the deconstruction